### PR TITLE
Push podspec to trunk in Release job

### PR DESCRIPTION
**Pull request by OdNairy**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog></changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

Push podspec to trunk in Release job

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -592,6 +592,9 @@ jobs:
           when: on_success
       - make-docs
       - open-docs-pull-requests
+      - run:
+          name: Publishe CocoaPods Podspec
```
Comment (macdrevx) [inline .circleci/config.yml:596]:
```suggestion
          name: Publish CocoaPods Podspec
```

**Review by macdrevx — COMMENTED** [review]
How will CI be configured as a pod owner?

**Comment by OdNairy** [comment]
> How will CI be configured as a pod owner?

It was configured with COCOAPODS_TRUNK_TOKEN environment variable

File: .circleci/config.yml (hunk 1 above)
Comment (OdNairy) [reply .circleci/config.yml]: Fixed

**Review by macdrevx — APPROVED** [review]

**Pull request merged**